infrastructure/templates/public-api/main.bicep

```diff
@@ -1,290 +1,206 @@
-    dbConnectionString: eesKeyVault.getSecret(databaseModule.outputs.connectionStringSecretName)
-    serviceBusConnectionString: eesKeyVault.getSecret(serviceBusFunctionQueueModule.outputs.connectionStringSecretName)
+    dbConnectionString: keyVault.getSecret(postgreSqlServerModule.outputs.connectionStringSecretName)
     tagValues: tagValues
+    applicationInsightsKey: applicationInsightsModule.outputs.applicationInsightsKey
+    subnetId: vNetModule.outputs.apiContainerAppSubnetRef
   }
   dependsOn: [
-    keyVaultModule
-    databaseModule
-    serviceBusFunctionQueueModule
+    postgreSqlServerModule
+    applicationInsightsModule
+    vNetModule
   ]
 }
 
-//Deploy Service Bus
-module serviceBusFunctionQueueModule 'components/serviceBusQueue.bicep' = {
-  name: 'serviceBusQueueDeploy'
+// Deploy data-processing Function.
+module dataProcessorFunctionAppModule 'application/dataProcessorFunctionApp.bicep' = {
+  name: 'functionAppDeploy-data-processor'
   params: {
     resourcePrefix: resourcePrefix
     location: location
-    namespaceName: namespaceName
-    queueName:queueName
-    keyVaultName: keyVaultModule.outputs.keyVaultName
+    functionAppName: 'data-processor'
+    storageAccountConnectionString: storageAccountConnectionString
+    dbConnectionString: keyVault.getSecret(postgreSqlServerModule.outputs.connectionStringSecretName)
     tagValues: tagValues
+    applicationInsightsKey: applicationInsightsModule.outputs.applicationInsightsKey
+    subnetId: vNetModule.outputs.dataProcessorSubnetRef
   }
   dependsOn: [
-    keyVaultModule
+    postgreSqlServerModule
+    applicationInsightsModule
+    vNetModule
   ]
 }
 
-//Deploy ETL Function
-module etlFunctionAppModule 'application/processorFunctionApp.bicep' = {
-  name: 'processorFunctionAppDeploy'
-  params: {
-    resourcePrefix: resourcePrefix
-    location: location
-    functionAppName: functionAppName
-    storageAccountConnectionString: eesKeyVault.getSecret(storageAccountModule.outputs.connectionStringSecretName)
-    dbConnectionString: eesKeyVault.getSecret(databaseModule.outputs.connectionStringSecretName)
-    serviceBusConnectionString: eesKeyVault.getSecret(serviceBusFunctionQueueModule.outputs.connectionStringSecretName)
-    tagValues: tagValues
-  }
-  dependsOn: [
-    keyVaultModule
-    databaseModule
-    storageAccountModule
-    serviceBusFunctionQueueModule
-  ]
-}
-
-
-//outputs
-output containerRegistryLoginServer string = containerRegistryModule.outputs.containerRegistryLoginServer
-output containerRegistryName string = containerRegistryModule.name
-output metadataDatabaseRef string = databaseModule.outputs.databaseRef
+// Outputs.
+output containerRegistryLoginServer string = containerRegistry.properties.loginServer
+output containerRegistryName string = containerRegistry.name
+output metadataDatabaseRef string = postgreSqlServerModule.outputs.databaseRef
 output managedIdentityName string = containerAppModule.outputs.managedIdentityName
```

Comment:
Will we have multiple managed identities in the future potentially?

It might be a good idea to make this output's name a bit more precise as it's a little too generic

Reply:
Yes I'd agree, and I've updated to be `apiContainerAppManagedIdentityName`.

I figure that down the line, this managed identity may be used both for doing the ACR image pulls *and* being used to connect to PSQL / Storage etc on the API Container App's behalf.